src/block/countdown/frontend-countdown.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready'
+import { date } from '@wordpress/date'
+
+const SECONDS = 1
+const SECONDS_IN_MINUTE = SECONDS * 60
+const SECONDS_IN_HOUR = SECONDS_IN_MINUTE * 60
+const SECONDS_IN_DAY = SECONDS_IN_HOUR * 24
+
+class StackableCountdown {
+	constructor( el ) {
+		this.el = el
+		this.blockId = el.getAttribute( 'data-block-id' )
+		this.countdownInterval
+	}
+
+	clearTimer = () => {
+		clearInterval( this.countdownInterval )
+	}
+
+	addZero = ( number, isDoubleDigit ) => {
+		if ( isDoubleDigit ) {
+			const a = number <= 0 ? 0 : number
+			return number < 10 ? '0' + a : number
+		}
+		return number <= 0 ? 0 : number
+	}
+
+	saveTimeRemaining = number => {
+		sessionStorage.setItem( this.blockId, number )
+	}
+
+	preCountDown = isDoubleDigit => {
+		const day = this.el.querySelector( '.stk-block-countdown__digit_day' )
+		const hour = this.el.querySelector( '.stk-block-countdown__digit_hour' )
+		const minute = this.el.querySelector( '.stk-block-countdown__digit_minute' )
+		const second = this.el.querySelector( '.stk-block-countdown__digit_second' )
+
+		const timer = sessionStorage.getItem( this.blockId )
+
+		const daysLeft = Math.floor( timer / SECONDS_IN_DAY )
+		const hoursLeft = Math.floor( ( timer % SECONDS_IN_DAY ) / SECONDS_IN_HOUR )
+		const minutesLeft = Math.floor( ( timer % SECONDS_IN_HOUR ) / SECONDS_IN_MINUTE )
+		const secondsLeft = Math.floor( ( timer % SECONDS_IN_MINUTE ) / SECONDS )
+
+		if ( day ) {
+			day.textContent = this.addZero( daysLeft, isDoubleDigit )
+		}
+
+		if ( hour ) {
+			hour.textContent = this.addZero( hoursLeft, isDoubleDigit )
+		}
+
+		if ( minute ) {
+			minute.textContent = this.addZero( minutesLeft, isDoubleDigit )
+		}
+
+		if ( second ) {
+			second.textContent = this.addZero( secondsLeft, isDoubleDigit )
+		}
+	}
+
+	countDown = ( countdownDate, isDoubleDigit, timezone, action, duration = 0, countdownType = 'dueDate', restartInterval = 0 ) => {
+		const day = this.el.querySelector( '.stk-block-countdown__digit_day' )
```

Comment:
Doing `querySelector` is slow. Instead of querying every time, you can do this at `init` and store the element so you won't have to query for it again.

Do this at `init`:
```js
this.dayEl = this.el.querySelector( '.stk-block-countdown__digit_day' )
```